core/src/main/java/jenkins/util/VirtualFile.java

```diff
@@ -714,7 +831,9 @@ private String computeRelativePathToRoot(){
                 current = current.getParent();
             }
 
-            return String.join(File.separator, relativePath) + File.separator;
+            // instead of File.separator that is specific to the master, the / has the advantage to be supported
+            // by either Windows AND Linux for the Path.toRealPath() used in isDescendant
+            return String.join("/", relativePath) + "/";
```

Comment:
this code is duplicated, shouldn't we encapsulate it for maintainability's sake?

Reply:
Good point, I put the method in the VirtualFile class directly. That will also remove the duplication of comment.